### PR TITLE
Update Drupal 8 demo buildkit build and create a Drupal 9 buildkit bu…

### PR DESCRIPTION
**Pull request by seamuslee001**
…ild as well for use as demo sites

ping @MikeyMJCO @totten @demeritcowboy

### CODE REVIEW

File: app/config/drupal9-demo/install.sh [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  drush8 -y updatedb
+popd >> /dev/null
+
+###############################################################################
+## Setup CiviCRM (config files, database tables)
+
+CIVI_DOMAIN_NAME="Demonstrators Anonymous"
+CIVI_DOMAIN_EMAIL="\"Demonstrators Anonymous\" <info@example.org>"
+CIVI_CORE="${CMS_ROOT}/vendor/civicrm/civicrm-core"
+CIVI_UF="Drupal8"
```
Comment (demeritcowboy) [inline app/config/drupal9-demo/install.sh:26]: 6 => Drupal6
7 => Drupal
8 => Drupal8
9 => Drupal8

I get it but ... yeah, drupal.

Comment (seamuslee001) [reply app/config/drupal9-demo/install.sh:26]: Yeh d8/9 are very similar so no real reason creating a separate UF for it

**Comment by demeritcowboy** [comment]
I'm noticing `drush dl` has been deprecated in D9. And I'm wondering should these be using drupal recommended project now, or at least for D9?

Also is exportui part of core now? Does it need the extension?

**Comment by MikeyMJCO** [comment]
Issue apply patch: 

> 537.patch:913: new blank line at EOF.
> +
> warning: 1 line adds whitespace errors.

**Comment by MikeyMJCO** [comment]
> I'm noticing `drush dl` has been deprecated in D9. And I'm wondering should these be using drupal recommended project now, or at least for D9?
> 
> Also is exportui part of core now? Does it need the extension?

Yeah, can we go the "proper" way of doing this please instead of trying to recreate D7 methodologies. Composer-first workflows for D8/D9.

**Comment by MikeyMJCO** [comment]
Okay so builds work:

[Drupal 8](https://d8.demo.bkonaz.mjco.uk/)

[Drupal 9](https://d9.demo.bkonaz.mjco.uk/)

**Comment by seamuslee001** [comment]
thanks @demeritcowboy @MikeyMJCO I have updated it to be all composer and also have re-added in those blocks @MikeyMJCO can you make sure that you have the latest drush8 there i think try running composer install in your buildkit dir

**Comment by demeritcowboy** [comment]
Thanks @seamuslee001. Is exportui needed anymore? It's been in core [since 5.23](https://github.com/civicrm/civicrm-core/pull/16335).

**Comment by MikeyMJCO** [comment]
> thanks @demeritcowboy @MikeyMJCO I have updated it to be all composer and also have re-added in those blocks @MikeyMJCO can you make sure that you have the latest drush8 there i think try running composer install in your buildkit dir

Issued a PR to update the Drush 8 version in composer.json. Will be required for D9 builds.

File: app/config/drupal8-demo/download.sh [hunk 2]
```diff
@@ -4,20 +4,29 @@
...
-  | sed "s;%%CMS_VERSION%%;${CMS_VERSION};" \
-  | sed "s;%%DISC_VERSION%%;${DISC_VERSION};" \
-  | sed "s;%%VOL_VERSION%%;${VOL_VERSION};" \
-  | sed "s;%%NG_PRFL_VERSION%%;${NG_PRFL_VERSION};" \
-  > "$MAKEFILE"
+mkdir "$WEB_ROOT"
+composer create-project drupal/recommended-project:"$CMS_VERSION" "$WEB_ROOT" --no-interaction
 
-drush8 -y make --working-copy "$MAKEFILE" "$WEB_ROOT"
+pushd "$WEB_ROOT" >> /dev/null
```
Comment (MikeyMJCO) [inline app/config/drupal8-demo/download.sh:16]: This could use the composer install function from civibuild.lib.sh like the d8prj builds do.

Comment (MikeyMJCO) [reply app/config/drupal8-demo/download.sh:16]: Same for D9 :-)

Comment (seamuslee001) [reply app/config/drupal8-demo/download.sh:16]: @MikeyMJCO is that not what is called in L25 there?

Comment (MikeyMJCO) [reply app/config/drupal8-demo/download.sh:16]: Yes, ignore me :-)

**Comment by MikeyMJCO** [comment]
D9 builds are still causing me issues:

```
+++ '[' '!' -d /opt/buildkit/build/d9.demo/vendor/civicrm/civicrm-core/bin -o '!' -d /opt/buildkit/build/d9.demo/vendor/civicrm/civicrm-core/CRM ']'
+++ cvutil_fatal 'Failed to locate valid civi root: /opt/buildkit/build/d9.demo/vendor/civicrm/civicrm-core'
+++ echo 'Failed to locate valid civi root: /opt/buildkit/build/d9.demo/vendor/civicrm/civicrm-core'
Failed to locate valid civi root: /opt/buildkit/build/d9.demo/vendor/civicrm/civicrm-core
+++ exit 90
```

**Comment by seamuslee001** [comment]
That is really strange
would have thought this would solve it https://github.com/civicrm/civicrm-buildkit/pull/537/files#diff-5250fce057ad705626e39890b8affc6aR25 that is what solves it for me but maybe something is screwy

**Comment by MikeyMJCO** [comment]
> That is really strange
> would have thought this would solve it https://github.com/civicrm/civicrm-buildkit/pull/537/files#diff-5250fce057ad705626e39890b8affc6aR25 that is what solves it for me but maybe something is screwy

Lemme grab a fresh copy of that file into my environment and try again - in case the patch went screwy

**Comment by seamuslee001** [comment]
Merging as ok by Mikey

**Pull request merged**